solutions/usaco-1061.mdx

```diff
@@ -0,0 +1,109 @@
+---
+id: usaco-1061
+source: USACO Bronze 2020 December
+title: Stuck in a Rut
+author: Ryan Chou
+---
+
+[Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_prob3_bronze_dec20.html)
+
+## Explanation
+
+Since $x, y \leq 10^9$, we can't simulate each time period. Instead, we should look for signs which allow two cows to intersect. 
+
+One important observation is that cows can't move backwards, so the only way for two cows to collide is if $n[x] > e[x] $ and $n[y] < e[y]$. 
+
+This ensures that the cows are positioned like this:
+
+![](USACO-dec20-stuckinarut-bronze/intersection_example.png)
+
+From here, we can track which cows will collide and how much grass they'll eat.
+
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+<LanguageSection>
+<PySection>
+  
+```py
+n = int(input())
+ncows = []
+ecows = []
+
+for i in range(n):
+	dir, x, y = input().split()
+	# Also append i since we have to return them
+	# in the same order they were given to us.
+	if dir == 'N':
+		ncows.append((int(x), int(y), i))
+	else:
+		ecows.append((int(x), int(y), i))
+
+# Sort north cows by x coordinates.
+ncows.sort()
+# Sort east cows by y coordinates.
+ecows.sort(key = lambda cow: cow[1])
+
+# Stores times which cows stop.
+stoppin = [None] * n
+# Checking every combination of cows.
+for ncow in ncows:
+	for ecow in ecows:
+		"""
+		The north cows can't move down, and east cows
+		can't move backwards, so a north cow's x coord must
```

Comment:
```suggestion
		can't move backwards, so a north cow's x coordinate must
```

Reply:
sometimes you can shorten, but i think typing out the full word is better here (change for the rest as well)

Reply:
👍